frontend/common/src/hooks/useRuntimeVariable.ts

```diff
@@ -0,0 +1,14 @@
+interface HasServerConfig {
+  __SERVER_CONFIG__?: Map<string, string>;
+}
+
+/**
+ * A hook to retrieve an environment variable value from the window object
+ */
+export const useRuntimeVariable = (name: string): string | undefined => {
```

Comment:
From what I can till, this function has no reason to be a hook. Hooks tend to rely on other hooks and therefore only work inside react components, but this isn't under that restriction. It would make more sense to me to live in the helpers folder.